### PR TITLE
Setup ssh-keys role with travis support

### PR DESCRIPTION
**Pull request by mcritchlow**
Changes proposed:
* Pull SSH keys from GitHub for populating accounts
* Use the [authorized_key](https://docs.ansible.com/ansible/latest/authorized_key_module.html) module to manage setting up keys
  * Note: The `comment` option requires Ansible 2.4
* Add Travis tests
  * Geerlinguy's idempotency stuff and docker setup
  * A role account test using `conan` with myself and @jhriv as users
* Add GitHub templates (PR, Issue, Contributing) to repo
* Add `README` documenting ways to use the role

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+2. Push commits to the feature branch, with appropriate test coverage (and tests passing).
+3. Prior to creating a Pull Request, make sure your branch is up to date with its parent branch (usually `develop`)
+```
+git checkout develop
+git pull --rebase
+git checkout <your-feature-branch>
+git rebase develop
+```
+4. Run your tests again.
+5. Run rubocop `rubocop` (unless you already have it integrated into your IDE/Editor) and fix reported issues. This can be done automatically in some cases with `rubocop -a`
```
Comment (jhriv) [inline .github/CONTRIBUTING.md]: Robocop? For this?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    - hosts: servers
+      roles:
+         - { role: ucsdlib.ansible-role-ssh-keys,
+             tags: ssh-keys,
+             ssh_keys_ad_account: 'conan',
+             ssh_keys_github_accounts: ['mcritchlow','jhriv','VivianChu']
+           }
+```
+
+Library Staff Map
```
Comment (jhriv) [inline README.md]: This map would go great in `vars/main.yml`.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (mcritchlow) [reply .github/CONTRIBUTING.md]: copy/paste.. you know we should probably come up with a setup of these templates for devops-y stuff

File: defaults/main.yml [hunk 3]
```diff
@@ -1,4 +1,6 @@
 ---
-
 # default user to apply the key to
-user: '{{ ansible_user_id }}'
+ssh_keys_ad_account: '{{ ansible_user_id }}'
+ssh_keys_github_accounts:
+  - '{{ ssh_keys_ad_account }}'
```
Comment (jhriv) [inline defaults/main.yml]: My concern here is when the AD username differs from the GitHub username, the wrong Gihub account will get queried. At best a 404, at worst undesired ssh keys will get applied.

If the mapping ends up in vars/main.yml, then that can be referenced

Comment (mcritchlow) [reply defaults/main.yml]: Yeah, sounds good. Thanks for catching that. I'll update it to use the hash instead.

**Comment by jhriv** [comment]
So. I was playing with the library staff mapping and came up with....

test.yml
``` syntax=yml
---

- name: Grab variables by dot notation
  hosts: localhost

  vars_files:
     - vars/main.yml

  vars:
     user: '{{ ansible_user_id }}'

  tasks:

     - name: provide by dots
       debug:
          msg: 'ad: {{ user }}; github: {{ ssh_map[user].github }}; Real name: {{ ssh_map[user].name }}'
```

vars/main.yml
``` syntax=yml
ssh_map:
   jhriv:
      name: John H. Robinson, IV
      github: jhriv
   mcritchlow:
      name: Matt Critchlow
      github: mcritchlow
   aflick:
      name: Allison Flick
      github: allisonflick
   drtrujillo:
      name: David Trujillo
      github: dt-ucsd
```

The realtime can be used in the comment section, as further in-place documentation.

**Comment by mcritchlow** [comment]
I can definitely add full names if you'd like. It would make the comments a bit better. But the `user` lookup is only going to work on people. For role accounts like `conan` that won't work. So that's why I was thinking when someone runs the playbook they're going to need to be more explicit about the user/role and the user(s) SSH keys being applied to the keys file. That ok with you?

**Comment by jhriv** [comment]
Having a way to override the default, or explicitly define in the case of a role account not being in the mapping is good.

Simple things simple, interesting things possible.

So - something like:
``` syntax=yml
with_items:
  - '{{ ad_usernames | default (ssh_map[user] }}'
```
could work seamlessly. `-e user=conan -e ad_usernames='["drtrujillo","jhriv","mcritchlow"]'` as a possible implementation

**Comment by mcritchlow** [comment]
@jhriv - Ok, I've added a couple commits to:
- use a map for dynamic lookups per your suggestions
- removed rubocop from the github templates

Let me know if there's anything else, thanks for the feedback.

**Review by jhriv — APPROVED** [review]
:shipit:

**Pull request merged**